### PR TITLE
Realtime testing opensplice C example

### PR DESCRIPTION
**Pull request by jacquelinekay**
Needs https://github.com/jacquelinekay/rttest/pull/1

This PR introduces three programs instrumented for real-time performance testing: a standalone publisher, a standalone subscriber, and an intra-process comms example. Each one uses the Opensplice C API.

### CODE REVIEW

File: realtime_ospl_c/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+mkdir c
+cd c
+OSPL_TMPL_PATH=/usr/etc/opensplice/idlpp idlpp -S -l c ../LargeMsg.idl
+echo "Generated C code under '`pwd`'"
+cd ..
+
+rm -fr build
+mkdir build
+cd build
+cmake .. -DCMAKE_INSTALL_PREFIX=~/local
```
Comment (dirk-thomas) [inline realtime_ospl_c/build.sh]: Should this be configured with `-DCMAKE_BUILD_TYPE=Release`?

**Comment by tfoote** [comment]
+1

File: realtime_ospl_c/build.sh (hunk 1 above)
Comment (jacquelinekay) [inline realtime_ospl_c/build.sh]: This build script was specialized for my build environment, so I decided to make it more flexible. In b09eb99e7f84b112efdb6863acce3282af4bb0a2 I modified this script to take pass arguments directly to the cmake command, so you can configure the build with `. build.sh -DCMAKE_BUILD_TYPE=Release`.

Comment (dirk-thomas) [inline realtime_ospl_c/build.sh]: Even better :wink:

**Comment by dirk-thomas** [comment]
LGTM but I haven't built / ran the programs.

**Comment by wjwwood** [comment]
+1

**Pull request merged**